docs/coding-guidelines/add-types-to-SerDe.md

```diff
@@ -0,0 +1,100 @@
+finding the method `ConvertArgsToBytes` and find out where in the switch that the code should be 
+implemented.
+
+This is how a byte array is written:
+
+```csharp
+
+    case double[][] argDoubleArrayArray:
+        SerDe.Write(destination, s_doubleArrayArrayTypeId);
+        SerDe.Write(destination, argDoubleArrayArray.Length);
+        foreach (double[] doubleArray in argDoubleArrayArray)
+        {
+            SerDe.Write(destination, doubleArray.Length);
+            foreach (double d in doubleArray)
+            {
+                SerDe.Write(destination, d);
+            }
+        }
+        break;
+
+```
+
+Typically the type identifier is written, then the length and then the data. Once you have written 
+the code that will perform the write you need to add a case to `GetTypeId` for your new identifier:
+
+```csharp
+
+    if (type == typeof(int[]) ||
+        type == typeof(long[]) ||
+        type == typeof(double[]) ||
+        type == typeof(double[][]) ||
+        typeof(IEnumerable<byte[]>).IsAssignableFrom(type) ||
+        typeof(IEnumerable<string>).IsAssignableFrom(type))
+    {
+        return s_arrayTypeId;
+    }
+```
+
+### 2. Derserialize this new type in JVM with read method
+
+In [SerDe.scala](https://github.com/dotnet/spark/blob/master/src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/api/dotnet/SerDe.scala) (This need to be changed in all Spark version 2.3.x, 2.4.x and 3.0.x). 
+
+In either the method `readTypedObject` or `readList` add a new case statement for your new type 
+identifier:
+
+```scala
+
+  case 'A' => readDoubleArrArr(dis)
+
+```
+
+Once you have read the type, you can implement your method to read the data you wrote in csharp, examples:
+
+```scala
+
+   def readDoubleArrArr(in: DataInputStream): Array[Array[Double]] = {
+    val len = readInt(in)
+    (0 until len).map(_ => readDoubleArr(in)).toArray
+  }
+
+```
+
+That should be everything you need to be able to send a new type from the CLR to the JVM.
+
+### 3. Serialize this new type in JVM with write method
+
+In src/main/scala/org/apache/spark/api/dotnet/SerDe.scala the function `writeObject` contains a switch for each type, a couple of examples:
+
+```scala
+
+ case "[[D" =>
+  writeType(dos, "list")
+  writeDoubleArrArr(dos, value.asInstanceOf[Array[Array[Double]]])
+
+```
+
+### 4. Add this new type in [JvmBridge.cs](https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs#L151) to read the value on CLR side
+
+The final step is to be able to read the value on the CLR side that the JVM side has written. Back in csharp find csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs add your type identifier to `CallJavaMethod` in the switch statement `switch (typeAsChar) //TODO: Add support for other types.` You will likely find that the type you want to implement should be implemented inside `ReadCollection`.
```

Comment:
Let's add the [sample code](https://github.com/GoEddie/spark/blob/1968f59c22df9808ab210fa7bf94ab5cdd48a9f2/src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs#L368) here as an example.